bundles/org.openhab.core.automation.module.script.scriptenginefactory.jython/src/main/java/org/openhab/core/automation/module/script/scriptenginefactory/jython/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.module.script.scriptenginefactory.jython;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.script.ScriptEngine;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.module.script.AbstractScriptEngineFactory;
+import org.openhab.core.automation.module.script.ScriptEngineFactory;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link ScriptEngineFactory} for Jython.
+ *
+ * @author Scott Rushworth - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = org.openhab.core.automation.module.script.ScriptEngineFactory.class)
+public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(JythonScriptEngineFactory.class);
+    private static final String SCRIPT_TYPE = "py";
+    private static javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
+
+    static {
+        String home = JythonScriptEngineFactory.class.getProtectionDomain().getCodeSource().getLocation().toString()
+                .replace("file:", "");
+        String openhabConf = System.getenv("OPENHAB_CONF");
+        System.setProperty("python.home", home);
+        System.setProperty("python.path", openhabConf + "/automation/lib/python");
```

Comment:
Currently, afaik, the `.../lib/<language>` path is a convention, rather than required in OH. This change switches it to make it required.
I have no problem with this, but wanted to raise it because I also embedded it in the JS plugin, but made it configurable. We should probably adopt the same approach; I'm happy to go with the simpler code of forcing it to a predefined value if you think this is the superior option.

Reply:
This is the directory structure convention suggested in the HL documentation, but for the JS HLs, it is required. However, I plan to add configuration options. Yesterday, I changed this to append the python.path, in case one was previously defined. This allows you to add third party libs in other locations. I also plan to add an option to set python.home, so that you can use this feature to get the HLs and use a custom Jython install (different version or full install).

Reply:
Something to note: in my JS implementation the module loader takes care of locating and loading the libs. This means that there is no more JS code attempting to do it, so it ceases to be required any more than it is in python. It basically brings it inline with (a) standard JS dev envs and (b) the way Python works. The equivalent of:
`from foo import bar`
is:
`const { bar} = require('foo')`